backend/src/routes/person.js

```diff
@@ -34,6 +34,128 @@ router.get("/:id/news", async(req, res) => {
     }
 });
 
+router.get("/:id/news/:newsId", async (req, res) => {
+  let id = req.params.id;
+  let newsId = req.params.newsId;
+  let data = {};
+    try {
+    const connection = await pool.getConnection();
+    const query = `
+      SELECT n.id, n.personId, n.title, n.content, n.published, n.teamId, Person.firstName, Person.lastName FROM News as n
+      JOIN TeamPlayer ON n.teamId = TeamPlayer.teamId
+      JOIN TeamStaff ON n.teamId = TeamStaff.teamId
+      JOIN Person ON n.personId = Person.id
+      WHERE (TeamPlayer.personId = ? OR TeamStaff.personId = ?)
+      AND n.id = ?
+      GROUP BY n.id
+    `;
+    const news = await connection.query(query, [id, id, newsId]);
+    data = news[0];
+    connection.release();
+    res.send(data);
+  } catch (err) {
+    console.error(err);
+    res.status(404);
+    res.send({
+      errorCode: "not_found",
+      errorMessage: "News article not found"
+    });
+  }
+});
+
+
+router.post("/:id/news", async (req, res) => {
+  let id = req.params.id;
+  let newsData = req.body;
+  try {
+    const connection = await pool.getConnection();
+    const query = `
+      INSERT INTO News (personId, teamId, title, content, published)
+      VALUES (?, ?, ?, ?, ?)
+    `;
+    const result = await connection.query(query, [id, newsData.teamId, newsData.title, newsData.content, newsData.published]);
+    const newsId = result.insertId;
+    connection.release();
+    res.send({
+      id: newsId,
+      message: "News article created successfully"
+    });
+  } catch (err) {
+    console.error(err);
+    res.status(500);
+    res.send({
+      errorCode: "internal_server_error",
+      errorMessage: "Failed to create news article"
+    });
+  }
+});
+
+router.put("/:id/news/:newsId", async (req, res) => {
+  let id = req.params.id;
+  let newsId = req.params.newsId;
+  let newsData = req.body;
+  console.log(`Edit news article ${newsId} for person with id: ${id}`);
+  try {
+    const connection = await pool.getConnection();
+    const query = `
+      UPDATE News
+      SET teamId = ?, title = ?, content = ?, published = ?
+      WHERE id = ?
+      AND personId = ?
+    `;
+    const result = await connection.query(query, [newsData.teamId, newsData.title, newsData.content, newsData.published, newsId, id]);
+    connection.release();
+    if (result.affectedRows === 0) {
+      res.status(404);
+      res.send({
+        errorCode: "not_found",
+        errorMessage: "News article not found"
+      });
+    } else {
+      res.send({
+        message: "News article updated successfully"
+      });
+    }
+  } catch (err) {
+    console.error(err);
+    res.status(500);
+    res.send({
+      errorCode: "internal_server_error",
+      errorMessage: "Failed to update news article"
+    });
+  }
+});
+
+
+router.delete("/:id/news/:newsId", async (req, res) => {
```

Comment:
Same as above, should be in a news route and only `/news/id`

Reply:
According to Jerome you cant use :id twice in same route.